magicctapipe/scripts/lst1_magic/coincident_events.py

```diff
@@ -0,0 +1,179 @@
+"""
+This scripts facilitates the usage of the script
+"lst1_magic_event_coincidence.py". This script is
+more like a "maneger" that organizes the analysis
+process by:
+1) Creating the bash scripts for looking for
+coincidence events between MAGIC and LST in each
+night.
+2) Creating the subdirectories for the coincident
+event files.
+
+
+Usage:
+$ python coincident_events.py
+
+"""
+
+import os
+import numpy as np
+import glob
+import yaml
+import logging
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+def configfile_coincidence(ids, target_dir):
+    
+    """
+    This function creates the configuration file needed for the event coincidence step
+    
+    Parameters
+    ----------
+    ids: list
+        list of telescope IDs
+    target_dir: str
+        Path to the working directory
+    """
+    
+    f = open(target_dir+'/config_coincidence.yaml','w')
+    f.write("mc_tel_ids:\n    LST-1: "+str(ids[0])+"\n    LST-2: "+str(ids[1])+"\n    LST-3: "+str(ids[2])+"\n    LST-4: "+str(ids[3])+"\n    MAGIC-I: "+str(ids[4])+"\n    MAGIC-II: "+str(ids[5])+"\n\n")
+    f.write('event_coincidence:\n    timestamp_type_lst: "dragon_time"  # select "dragon_time", "tib_time" or "ucts_time"\n    window_half_width: "300 ns"\n')
+    f.write('    pre_offset_search: true\n')  
+    f.write('    n_pre_offset_search_events: 100\n')  
+    f.write('    time_offset:\n        start: "-10 us"\n        stop: "0 us"\n')  
+    f.close()
+    
+
+def linking_lst(target_dir, LST_runs):
+    
+    """
+    This function links the LST data paths to the working directory. This is a preparation step required for running lst1_magic_event_coincidence.py 
+    
+    Parameters
+    ----------
+    target_dir: str
+        Path to the working directory
+    LST_runs: matrix of strings
+        This matrix is imported from config_general.yaml and tells the function where to find the LST data and link them to our working directory
+    """
+    
+    
+    coincidence_DL1_dir = target_dir+"/DL1/Observations"
+    if not os.path.exists(coincidence_DL1_dir+"/Coincident"):
+        os.mkdir(f"{coincidence_DL1_dir}/Coincident")
+    
+    for i in LST_runs:
+        lstObsDir = i[0].split("_")[0]+i[0].split("_")[1]+i[0].split("_")[2]
+        inputdir = f'/fefs/aswg/data/real/DL1/{lstObsDir}/v0.9/tailcut84'
```

Comment:
maybe the version of LST data should be also an argument to be passed somewhere. I do not expect people to change the source code to access a different version of LST data.

Reply:
We can add it in the config_general (where we already pass the Simtel version)